File: lab-2.3-ByLiza/C#/TextContainer.cs
using System.Collections.Generic;
using System.Linq;

public class TextContainer
{
    private List<MyString> lines = new();

    public void AddLine(string input)
    {
        lines.Add(new MyString(input));
    }

    public void RemoveLine(int index)
    {
        if (index >= 0 && index < lines.Count)
            lines.RemoveAt(index);
    }

    public void Clear()
    {
        lines.Clear();
    }

    public void CapitalizeAll()
    {
        foreach (var line in lines)
        {
            line.CapitalizeWords();
        }
    }

    public string GetKeyFromLine(int index)
    {
        if (index >= 0 && index < lines.Count)
            return lines[index].GenerateKey();
        return "";
    }

    public int CountLinesOfLength(int length)
    {
        return lines.Count(line => line.Length == length);
    }

    public void PrintAll()
    {
        foreach (var line in lines)
        {
            System.Console.WriteLine(line);
        }
    }
}

File: lab-2.3-ByLiza/C++/TextContainer.cpp
#include "TextContainer.h"
#include <iostream>

void TextContainer::AddLine(IKeyGenerator* input) {
    lines.push_back(input);
}

void TextContainer::RemoveLine(size_t index) {
    if (index < lines.size())
        lines.erase(lines.begin() + index);
}

void TextContainer::Clear() {
    lines.clear();
}

void TextContainer::CapitalizeAll() {
    for (auto& line : lines) {
        line->CapitalizeWords();
    }
}

std::string TextContainer::GetKeyFromLine(size_t index) const {
    if (index < lines.size())
        return lines[index]->GenerateKey();
    return "";
}

int TextContainer::CountLinesOfLength(size_t length) const {
    int count = 0;
    for (const auto& line : lines) {
        if (line->GetContent().length() == length)
            ++count;
    }
    return count;
}

void TextContainer::PrintAll() const {
    for (const auto& line : lines) {
        std::cout << line->GetContent() << std::endl;
    }
}